core/src/main/java/org/jdbi/v3/core/statement/Batch.java

```diff
@@ -77,16 +78,23 @@ public Batch add(String sql) {
                 throw new UnableToExecuteStatementException("Unable to configure JDBC statement", e, getContext());
             }
 
+            SqlLogger sqlLogger = getConfig(SqlStatements.class).getSqlLogger();
             try {
-                final long start = System.nanoTime();
+                getContext().setExecutionMoment(Instant.now());
+                sqlLogger.logBeforeExecution(getContext());
+
                 final int[] rs = stmt.executeBatch();
-                final long elapsedTime = System.nanoTime() - start;
-                LOG.trace("] executed in {}ms", elapsedTime / 1000000L);
-                // Null for statement, because for batches, we don't really have a good way to keep the sql around.
-                getConfig(SqlStatements.class).getTimingCollector().collect(elapsedTime, getContext());
-                return rs;
 
+                getContext().setCompletionMoment(Instant.now());
+                sqlLogger.logAfterExecution(getContext());
+
+                LOG.trace("] executed in {}ms", getContext().getElapsedTime(ChronoUnit.MILLIS));
```

Comment:
I think the addition of `SqlLogger` means that these trace statements could go away.